libminifi/include/utils/IntegrationTestUtils.h

```diff
@@ -0,0 +1,66 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <utility>
+#include <string>
+#include <vector>
+
+#include "../../../libminifi/test/TestBase.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+template <class Rep, class Period, typename Fun>
+bool verifyEventHappenedInPollTime(const std::chrono::duration<Rep, Period>& wait_duration, Fun&& check) {
+  std::chrono::system_clock::time_point wait_end = std::chrono::system_clock::now() + wait_duration;
+  do {
+    if (std::forward<Fun>(check)()) {
+      return true;
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  } while (std::chrono::system_clock::now() < wait_end);
+  return false;
+}
+
+template <class Rep, class Period, typename ...String>
+bool verifyLogLinePresenceInPollTime(const std::chrono::duration<Rep, Period>& wait_duration, String... patterns) {
+  // This helper is to be removed once we upgrade to support gcc 4.9+ only
```

Comment:
what is the gcc issue here?  https://godbolt.org/z/hbMvss compiles fine on gcc 4.8.5 (and even on 4.7.4), at least in this sandbox

Reply:
Ideally you should not need to unpact the arguments into a new vector. However, unpacking does not properly work on gcc 4.8.5 when defining a variadic check.

Reply:
Sorry for being dense, but if I find this comment later, after we have upgraded to gcc >= 4.9 only, I'll have no idea how I am supposed to rewrite it.  Maybe include the better (post gcc-4.9) version of the code in the comment?

Reply:
Sure, I will improve the comment with details :)

Reply:
Updated the comment.